### PR TITLE
Simplify DB upserts

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #240

### CODE REVIEW

**Review by LunarX — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/MessageController.kt [hunk 1]
```diff
@@ -41,29 +40,32 @@ object MessageController {
...
-            deleteMessages(deletableMessages)
+            deleteMessages(outdatedMessages)
+        }
+    }
+
+    private fun getOutdatedMessages(realmMessages: List<Message>, apiMessages: List<Message>): List<Message> {
+        return realmMessages.filter { realmMessage ->
+            apiMessages.none { apiMessage -> apiMessage.uid == realmMessage.uid }
+        }
+    }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/MessageController.kt:64]: Simplify this with a realm query

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/MessageController.kt [hunk 2]
```diff
@@ -41,29 +40,32 @@ object MessageController {
     //endregion
 
     //region Edit data
-    fun update(apiMessages: List<Message>, thread: Thread) {
-
-        // Get current data
-        Log.d(RealmDatabase.TAG, "Messages: Get current data")
-        val realmMessages = thread.messages
+    fun update(realmMessages: List<Message>, apiMessages: List<Message>) {
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/MessageController.kt:43]: This method need a refactor, comments are useless

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/MessageController.kt [hunk 3]
```diff
@@ -41,29 +40,32 @@ object MessageController {
...
+            Log.d(RealmDatabase.TAG, "Messages: Get outdated data")
+            val outdatedMessages = getOutdatedMessages(realmMessages, apiMessages)
 
-        RealmDatabase.mailboxContent.writeBlocking {
             // Save new data
             Log.d(RealmDatabase.TAG, "Messages: Save new data")
-            apiMessages.forEach { apiMessage ->
-                if (!apiMessage.isManaged()) copyToRealm(apiMessage, UpdatePolicy.ALL)
-            }
+            insertNewData(apiMessages)
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/MessageController.kt:52]: Can we replace this with `copyListToRealm` ?

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/MessageController.kt (hunk 1 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/MessageController.kt:64]: Blocked by https://github.com/realm/realm-kotlin/issues/591

**Review by JorisBodin — APPROVED** [review]

**Pull request merged**